### PR TITLE
Add WPP Flags to support DoTraceMessage in UVC

### PR DESCRIPTION
**Pull request by JoelCorley**

### CODE REVIEW

File: camera/Tracing/lib/TraceFn.ps1 [hunk 1]
```diff
@@ -234,6 +238,32 @@ function Prepare-Target {
...
         }
 
+        # If target is Container prepare container sharing etc.
+        if($TargetType -eq [Tracing.TargetType]::Container) {
+            $testpath = cmdd dir c:\windows\system32\hcsdiag.exe -HideOutput
+            if( $testpath.ExitCode -ne 0) {
+                Write-Error "[Prepare-Target] TargetType: $TargetType is wrong, target does not have necessary tools"
+            }
+            
+            # We require that container is already running and obtain it id guid
```
Comment (saredd) [inline camera/Tracing/lib/TraceFn.ps1]: nit: its

File: camera/Tracing/lib/TraceFn.ps1 [hunk 2]
```diff
@@ -234,6 +238,32 @@ function Prepare-Target {
...
+            $containerList = execd hcsdiag list -HideOutput
+            if( ($containerList.Output -eq $null) -or ($containerList.Output.Split(",")[1].trim() -ne "Running"))
+            {
+                Write-Error "[Prepare-Target] TargetType: $TargetType there is no container running on target"
+            }
+            $script:ContainerId = ($containerList.Output.split(",")[2].trim())
+
+            # Id is GUID so lenght should be 36
```
Comment (saredd) [inline camera/Tracing/lib/TraceFn.ps1]: nit: typo

File: camera/Tracing/lib/Utils.psm1 [hunk 3]
```diff
@@ -427,6 +427,41 @@ function Test-DeviceConnected {
...
+        Get-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+    }
+
+    process {
+        $testpath = cmdd dir c:\windows\system32\hcsdiag.exe -HideOutput
+        if( $testpath.ExitCode -ne 0) {
+            return $false
+        }
+        
+        # We require that container is already running and obtain it id guid
```
Comment (saredd) [inline camera/Tracing/lib/Utils.psm1]: same: its

File: camera/Tracing/lib/Utils.psm1 [hunk 4]
```diff
@@ -427,6 +427,41 @@ function Test-DeviceConnected {
...
+        $containerList = execd hcsdiag list -HideOutput
+        if( ($containerList.Output -eq $null) -or ($containerList.Output.Split(",")[1].trim() -ne "Running")) {
+            return $false
+        }
+        $script:ContainerId = ($containerList.Output.split(",")[2].trim())
+
+        # Id is GUID so lenght should be 36
```
Comment (saredd) [inline camera/Tracing/lib/Utils.psm1]: same

File: camera/Tracing/lib/Utils.psm1 [hunk 5]
```diff
@@ -672,6 +730,10 @@ function RmDir-Target {
                 RmDir-Device $Path /S /Q
                 break
             }
+            ([Tracing.TargetType]::Container) {
```
Comment (saredd) [inline camera/Tracing/lib/Utils.psm1]: nit: doesn't matter too much but the order of the case statements sometimes has container last (matching enum ordering) and sometimes after Tshell (which the case is usually identical to)

File: camera/Tracing/lib/TraceFn.ps1 (hunk 1 above)
Comment (JoelCorley) [reply camera/Tracing/lib/TraceFn.ps1]: Sorry.  This wasn't part of my change.  I'm not going to fix this.

File: camera/Tracing/lib/TraceFn.ps1 (hunk 2 above)
Comment (JoelCorley) [reply camera/Tracing/lib/TraceFn.ps1]: Sorry.  This wasn't part of my change.  I'm not going to fix this.

File: camera/Tracing/lib/Utils.psm1 (hunk 4 above)
Comment (JoelCorley) [reply camera/Tracing/lib/Utils.psm1]: Sorry.  This wasn't part of my change.  I'm not going to fix this.

File: camera/Tracing/lib/Utils.psm1 (hunk 3 above)
Comment (JoelCorley) [reply camera/Tracing/lib/Utils.psm1]: Sorry.  This wasn't part of my change.  I'm not going to fix this.

**Comment by JoelCorley** [comment]
My change is just the last change block in camera/Tracing/wprp/Trace_Multimedia.wprp.  See: https://github.com/microsoft/busiotools/pull/67/files#diff-24641d04ea1ecbca177446b4e3d9da8f

Since this is already checked into the internal repo I'm going to abandon this PR and let Kasperi resolve any remaining issue and publish my change with his externally as my change is not urgent.

**Comment by JoelCorley** [comment]
Abandoning PR?

**Pull request closed without merging**